include/Particle_IC_Constructor.h

```diff
@@ -0,0 +1,120 @@
+#ifndef __PARTICLE_IC_CONSTRUCTOR_H__
+#define __PARTICLE_IC_CONSTRUCTOR_H__
+#include "GAMER.h"
+#include "TestProb.h"
+#include"vector"
+#include<iostream>
+#include<fstream>
+#include<sstream>
+using namespace std;
+
+//gsl library
+#include <gsl/gsl_integration.h>
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_math.h>
+#include <gsl/gsl_roots.h>
+
+
+typedef struct Filename_Parameter{
+   int Cloud_Num;
+
+   vector<string> Params_Filenames;
+
+}FP;
+typedef struct Physical_Parameter{
+   int Cloud_Num;
+
+   string Params_Filenames;
+
+   char   Cloud_Type[MAX_STRING];
+   char   Density_Table_Name[MAX_STRING];
+   int    OPT_EXT_POT;
+   char   ExtPot_Table_Name[MAX_STRING];
+   int      Cloud_RSeed;
+   double   Cloud_Rho0;
+   double   Cloud_R0;
+   double   Cloud_MaxR;
+   double*  Cloud_Center;
+   double*  Cloud_BulkVel;
+   double   Cloud_Einasto_Power_Factor;
+   double   Cloud_Par_Num_Ratio;
```

Comment:
To make this class more stand-alone, perhaps we should store `Cloud_Par_Num` instead of `Cloud_Par_Num_Ratio` for each cloud. We can still load `Cloud_Par_Num_Ratio` from the cloud runtime table since that's more convenient. But then we immediately convert `Cloud_Par_Num_Ratio` to `Cloud_Par_Num`.